q2_fmt/_util.py

```diff
@@ -1063,21 +1067,42 @@ def _create_duplicated_recip_table(mismatched_df, recip_df):
             'Feature2': [0, 1, 0],
             'Feature3': [0, 0, 1]}).set_index('id')
 
-    >>> _create_duplicated_recip_table(mismatched_df, recip_df)
+    >>> donor_df = pd.DataFrame({
+            'id': ['donor1', 'donor2', 'donor3'],
+            'Feature1': [1, 0, 0],
+            'Feature2': [0, 1, 0],
+            'Feature3': [0, 0, 1]}).set_index('id')
 
-    pd.DataFrame({
-            'id': ["sample1", "sample1", "sample2", "sample2",
-                   "sample3", "sample3"],
+    >>> _create_duplicated_recip_table(mismatched_df, recip_df, donor_df)
+
+    recip_df = pd.DataFrame({
+            'id': ["sample1..1", "sample1..2", "sample2..3", "sample2..4",
+                   "sample3..4", "sample3..6"],
             'Feature1': [1, 1, 0, 0, 0, 0],
             'Feature2': [0, 0, 1, 1, 0, 0],
             'Feature3': [0, 0, 0, 0, 1, 1]}).set_index('id')
-    """
-    duplicated_table = \
-        mismatched_df.merge(recip_df, left_index=True, right_index=True,
-                            how='left').drop(mismatched_df.columns, axis=1)
-    return duplicated_table
-
 
+        pd.DataFrame({
+            'id': ["donor2..1", "donor3..2", "donor1..3", "donor3..4",
+                   "donor1..5", "donor3..6"],
+            'Feature1': [1, 1, 0, 0, 0, 0],
+            'Feature2': [0, 0, 1, 1, 0, 0],
+            'Feature3': [0, 0, 0, 0, 1, 1]}).set_index('id')
+    """
+    simulate_recip_df = pd.DataFrame([], columns=recip_df.columns)
+    simulate_donor_df = pd.DataFrame([], columns=donor_df.columns)
+    count = 1
+    for recip, donor_row in simulated_mismatched_df.iterrows():
```

Comment:
`for idx, (recip, donor_row) in enumerated(simulated_mismatched_df.iterrrows()):`

Reply:
Actually life will be easier if there's a series instead of a df, then convert todict, then call .items to get tuples.